tidal-parse/src/Sound/Tidal/Parse.hs

```diff
@@ -708,6 +750,9 @@ instance Parse (Pattern Double -> (Pattern a -> Pattern a) -> Pattern a -> Patte
     $(fromTidal "someCyclesBy") <|>
     $(fromTidal "plyWith")
 
+instance Parse (Pattern T.Note -> (Pattern a -> Pattern a) -> Pattern a -> Pattern a) where
+  parser = parser
```

Comment:
This looks strange, and may create the `<<loop>>` when called? Some `fmap T.Note` missing?

Reply:
Yeah, that's definitely infinite/empty recursion. If you have, for example an example of a function with that type you could do: parser = $(fromTidal "thatFunction")

Reply:
uhm... ok... and what if I haven't? at the moment only params "note", "midinote" and "up".
I tried also with `empty`, but, no luck

Reply:
If there's no way of making that type (Pattern Note -> (Pattern a -> Pattern a) -> Pattern a) in Tidal then you don't need to make a parser/Parse instance for it at all. I can comment more if you can give some examples of actual code you might want to parse.

Reply:
What exactly *is* the failing test here? (Is it visible in Travis, or how do I reproduce?)

Reply:
I have to tell you... I didn't need to parse anything, I was just trying to fix compilation errors and failing tests, but I'm not fully aware of the behavior of tidal-parse code, so any help would be appreciated :)

Reply:
@jwaldmann here's the errors, at the end of the log:
https://travis-ci.org/github/tidalcycles/Tidal/jobs/732245834

Reply:
I will look at this later this afternoon. Still it would help to have some examples of this Note stuff in action - being mentioned here is the first I've heard of any of this...

Reply:
Note is just a sort of alias of Double, born to solve this issue: https://github.com/tidalcycles/Tidal/issues/702

Reply:
I see two places where you have parser = parser, which creates an infinite loop. If there is really no expression that evaluates to the types (at those places where you currently have parser = parser) then those instances/parsers (all of them) shouldn't be there at all; moreover, any pathways in other instances/parsers that reference those will also need to be deleted. I would expect it would "work" if both/all were defined as empty - perhaps you had defined one of them as empty and left the other in for the test?  Anyway, can I suggest that we start by removing pathways that don't correspond to any actual Tidal code - and then if it still causes problems, I'll look closer? (As the person usually maintaining this slightly complicated structure, I don't want to deal with the extra cognitive load of code that doesn't do anything!)